src/app/components/audio-recordings/pages/download/download.component.ts

```diff
@@ -218,151 +66,64 @@ class DownloadAudioRecordingsComponent
       return;
     }
     this.models = models;
+    this.model.projects = this.project ? [this.project.id] : [];
+    this.model.regions = this.region ? [this.region.id] : [];
+    this.model.sites = this.site ? [this.site.id] : [];
     this.updateHref();
-
-    const hideProjects = this.config.settings.hideProjects;
-    this.fields = [];
-
-    if (!hideProjects) {
-      if (this.project) {
-        this.fields.push(this.getProjectFormInput());
-      } else {
-        // TODO Future work, read project list and make typeahead
-      }
-    }
-
-    if (this.region) {
-      this.fields.push(this.getRegionFormInput(hideProjects));
-    } else {
-      // TODO Future work, read regions/sites list and make typeahead
-    }
-
-    if (this.site) {
-      this.fields.push(this.getSiteFormInput(hideProjects));
-    } else {
-      // TODO Future work, read sites list and make typeahead
-    }
-
-    this.fields.push(...this.dateSelectionRange());
   }
 
   public get project(): Project {
     return this.models[projectKey] as Project;
   }
+
   public get region(): Region {
     return this.models[regionKey] as Region;
   }
+
   public get site(): Site {
     return this.models[siteKey] as Site;
   }
 
-  private getProjectFormInput(): FormlyFieldConfig {
-    return {
-      key: "projectId",
-      type: "select",
-      defaultValue: this.project.id,
-      templateOptions: {
-        label: "Project",
-        // TODO Allow user to change selected site
-        disabled: true,
-        options: [{ value: this.project.id, label: this.project.name }],
-      },
-    };
-  }
-
-  private getRegionFormInput(hideProject: boolean): FormlyFieldConfig {
-    return {
-      key: "regionId",
-      type: "select",
-      defaultValue: this.region.id,
-      templateOptions: {
-        label: hideProject ? "Project" : "Site",
-        // TODO Allow user to change selected site
-        disabled: true,
-        options: [{ value: this.region.id, label: this.region.name }],
-      },
-    };
-  }
-
-  private getSiteFormInput(hideProject: boolean): FormlyFieldConfig {
-    return {
-      key: "siteId",
-      type: "select",
-      defaultValue: this.site.id,
-      templateOptions: {
-        label: hideProject ? "Site" : "Point",
-        // TODO Allow user to change selected site
-        disabled: true,
-        options: [{ value: this.site.id, label: this.site.name }],
-      },
-    };
+  public setRecordingStartedAfterDate(event: Event): void {
+    this.model.recordingStartedAfter = (event.target as HTMLInputElement).value;
```

Comment:
I believe ngForm has an event for whenever the form is updated.

Actually ngForm supports two-way data binding... why aren't you using ngForm?